controllers/topicPage/TopicPage.js

```diff
@@ -18,43 +18,70 @@ class TopicPage {
     this.getTopicPageById = this.getTopicPageById.bind(this);
   }
 
+  async getFeedTopicPages(payload) {
+    const topicPages = await this._getStreamService.getTopicPages(
+      payload.id,
+      payload.limit,
+      payload.offset
+    );
+    let newTopicPagesWithBlock = new BlockServices(
+      payload.listBlockUser,
+      payload.listBlockDomain,
+      payload.listPostAnonymous
+    ).getHasBlock(topicPages);
+    let results = await filterFeeds(payload.userId, newTopicPagesWithBlock, payload.id);
+    return results;
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `getFeedTopicPages` method seems to be well implemented. It fetches topic pages, applies block list filters, and further filters feeds based on user ID. However, it's important to ensure that the `filterFeeds` function handles any exceptions that might occur during its execution.

```diff
  async getFeedTopicPages(payload) {
    const topicPages = await this._getStreamService.getTopicPages(
      payload.id,
      payload.limit,
      payload.offset
    );
    let newTopicPagesWithBlock = new BlockServices(
      payload.listBlockUser,
      payload.listBlockDomain,
      payload.listPostAnonymous
    ).getHasBlock(topicPages);
    let results;
    try {
      results = await filterFeeds(payload.userId, newTopicPagesWithBlock, payload.id);
    } catch (error) {
      console.error('Error filtering feeds:', error);
      throw error;
    }
    return results;
  }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->